plugins/modules/ec2_launch_template.py

```diff
@@ -329,7 +346,32 @@
     - A set of key-value pairs to be applied to resources when this Launch Template is used.
     - "Tag key constraints: Tag keys are case-sensitive and accept a maximum of 127 Unicode characters. May not begin with I(aws:)"
     - "Tag value constraints: Tag values are case-sensitive and accept a maximum of 255 Unicode characters."
+    - This field is deprecated and will be removed in a release after 2026-12-01, use O(tag_specifications) instead.
     aliases: ['resource_tags']
+  tag_specifications:
+    description:
+    - The tags to apply to the resources when this Launch template is used.
+    type: list
+    elements: dict
+    version_added: 9.0.0
+    suboptions:
+      resource_type:
+        description:
+          - The type of resource to tag.
+          - If the instance does not include the resource type that you specify, the instance launch fails.
+        type: str
+        default: instance
+        choices:
```

Comment:
Any reason we cannot support more resource_types?

@tremble @markuman Any thoughts on this parameter?

Reply:
@alinabuzachis we cannot support more resource types because this is a limitation on the API and the definition of the launch template. See https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2/client/create_launch_template.html

Reply:
In the API spec I see a  longer list of supported resources. Am I wrong?

Reply:
@alinabuzachis you see a long list in the request syntax, but you need to refer to the parameter description
```
Valid Values lists all resource types for Amazon EC2 that can be tagged. When you create a launch template, you can specify tags for the following resource types only: instance | volume | network-interface | spot-instances-request. If the instance does not include the resource type that you specify, the instance launch fails. For example, not all instance types include a volume.
```

Reply:
Ah, ok, I see now.